### PR TITLE
Revert back to the group serialization for ProtVer for PParams

### PR DESCRIPTION
**Pull request by lehins**
# Description

Revert back to the group serialization for ProtVer for PParams. This PR fixes the same problem with PParams serialization that was fixed in #3297, but it fixes in a slightly more backwards compatible manner. Namely it, as before #3297, unrolls ProtocolVersion into a list, except it properly adjusts the number of elements in the list.

Both Alonzo and Babbage PParams were affected and fixed in this PR.

Implementation in this PR was discussed here https://github.com/CardanoSolutions/ogmios/issues/314#issuecomment-1554824499

Note the new version in the CHANGELOG, this is because this fix is getting backported to 1.2.1 versions in #3441 

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/PParams.hs [hunk 1]
```diff
@@ -278,90 +282,63 @@ instance Crypto c => EraGovernance (AlonzoEra c) where
...
+      <> encCBOR appKeyDeposit
+      <> encCBOR appPoolDeposit
+      <> encCBOR appEMax
+      <> encCBOR appNOpt
+      <> encCBOR appA0
+      <> encCBOR appRho
+      <> encCBOR appTau
+      <> encCBOR appD
+      <> encCBOR appExtraEntropy
+      <> encCBORGroup appProtocolVersion
```
Comment (JaredCorduan) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/PParams.hs:301]: :+1:

**Review by JaredCorduan — APPROVED** [review]
LGTM

**Comment by JaredCorduan** [comment]
hmm, a golden test is failing in CI. There are no protocol parameters in the script context, so I'm confused.

Edit: maybe I get it, the golden test needs to deserialize the entire transaction so it can create the script context. and a transaction can have a ppup.

**Comment by lehins** [comment]
> maybe I get it, the golden test needs to deserialize the entire transaction so it can create the script context. and a transaction can have a ppup.

Nah, its not that. PParams are generated because they are needed for TxInfo translation. But they also need to be stored somewhere. This one is easy, all I need is rerun the generator for txInfo translation golden tests. 

FYI, this problem does not apply to #3441

**Comment by JaredCorduan** [comment]
> Nah, its not that. PParams are generated because they are needed for TxInfo translation. 

oh, you mean to get a major protocol version?

**Comment by lehins** [comment]
> oh, you mean to get a major protocol version?

Yes. Here https://github.com/input-output-hk/cardano-ledger/blob/040b3a4b3993337dc07064e28e988ce99a1aef99/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs#L532

**Pull request merged**